tests/testthat/test-driver-sql-server.R

```diff
@@ -79,271 +79,258 @@ test_that("SQLServer", {
-    expect_equal(res, ir)
-
-    # Test: We can enumerate schemas out of catalog ( #527 )
-    # Part 1: Make sure we can see the schema we created in the
-    # current catalog.
-    res <- odbcConnectionSchemas(con)
-    expect_true("testSchema" %in% res)
-    # Part 2: Make sure we don't see that schema in the tempdb
-    # listing ( out of catalog schema listing )
-    res <- odbcConnectionSchemas(con, catalog_name = "tempdb")
-    # Should, at least, have INFORMATION_SCHEMA and sys
-    expect_true(length(res) > 1)
-    expect_false("testSchema" %in% res)
+test_that("works with schemas (#197)", {
+  con <- test_con("SQLSERVER")
+  dbExecute(con, "DROP SCHEMA IF EXISTS testSchema")
+  dbExecute(con, "CREATE SCHEMA testSchema")
+  on.exit({
+    dbExecute(con, "DROP TABLE testSchema.iris")
+    dbExecute(con, "DROP SCHEMA testSchema")
   })
 
-  local({
-    # SQLServer works with dbAppendTable (#215)
-    con <- DBItest:::connect(DBItest:::get_default_context())
+  ir <- iris
+  ir$Species <- as.character(ir$Species)
+
+  table_id <- Id(schema = "testSchema", table = "iris")
+  dbWriteTable(conn = con, name = table_id, value = ir)
+  dbWriteTable(conn = con, name = table_id, value = ir, append = TRUE)
+
+  res <- dbReadTable(con, table_id)
+  expect_equal(res, rbind(ir, ir))
+
+  dbWriteTable(conn = con, name = table_id, value = ir, overwrite = TRUE)
+  res <- dbReadTable(con, table_id)
+  expect_equal(res, ir)
+
+  # Test: We can enumerate schemas out of catalog ( #527 )
+  # Part 1: Make sure we can see the schema we created in the
+  # current catalog.
+  res <- odbcConnectionSchemas(con)
+  expect_true("testSchema" %in% res)
+  # Part 2: Make sure we don't see that schema in the tempdb
+  # listing ( out of catalog schema listing )
+  res <- odbcConnectionSchemas(con, catalog_name = "tempdb")
+  # Should, at least, have INFORMATION_SCHEMA and sys
+  expect_true(length(res) > 1)
+  expect_false("testSchema" %in% res)
+})
 
-    ir <- iris
-    ir$Species <- as.character(ir$Species)
+test_that("works with dbAppendTable (#215)", {
+  con <- test_con("SQLSERVER")
 
-    dbWriteTable(con, "iris", ir)
-    on.exit(dbRemoveTable(con, "iris"))
+  ir <- iris
+  ir$Species <- as.character(ir$Species)
 
-    dbAppendTable(conn = con, name = "iris", value = ir)
+  tbl <- local_table(con, "iris", ir)
+  dbAppendTable(con, tbl, ir)
 
-    res <- dbReadTable(con, "iris")
-    expect_equal(res, rbind(ir, ir))
-  })
+  res <- dbReadTable(con, tbl)
+  expect_equal(res, rbind(ir, ir))
+})
 
-  local({
-    # Subseconds are retained upon insertion (#208)
-    con <- DBItest:::connect(DBItest:::get_default_context())
+test_that("Subseconds are retained upon insertion (#208)", {
+  con <- test_con("SQLSERVER")
+  data <- data.frame(time = Sys.time())
+  tbl <- local_table(con, "time", data)
 
-    data <- data.frame(time = Sys.time())
-    dbWriteTable(con, "time", data, field.types = list(time = "DATETIME"), overwrite = TRUE)
```

Comment:
Is this `field.types = list(time = "DATETIME")` needed to retain the utility of the test?

Reply:
I don't think so; POSIXct are already detected as date time